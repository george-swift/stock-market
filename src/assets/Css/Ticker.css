.wrap-ticker {
  top: 0;
  width: 100%;
  overflow: hidden;
  padding-left: 100%;
  background-color: rgba(255, 255, 255, 0.4);
}

.wrap-ticker,
.ticker {
  box-sizing: content-box;
  height: 1.85rem;
}

@keyframes ticker {
  0% {
    transform: translate3d(0, 0, 0);
    visibility: visible;
  }

  100% {
    transform: translate3d(-100%, 0, 0);
  }
}

.ticker {
  display: inline-block;
  line-height: 1.85rem;
  white-space: nowrap;
  padding-right: 100%;
  animation: ticker 125s linear infinite;
}

.ticker__item {
  display: inline-block;
  padding: 0 1rem;
  font-size: 0.85rem;
}

.ticker:hover {
  animation-play-state: paused;
}
