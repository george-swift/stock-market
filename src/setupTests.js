/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });
