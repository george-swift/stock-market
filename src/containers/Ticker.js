import '../assets/Sass/Ticker.scss';
import { useSelector } from 'react-redux';
import { getListing } from '../selectors';

const Ticker = () => {
  const companies = useSelector((state) => getListing(state));

  return (
    <div className="overflow-hidden">
      <div className="wrap-ticker">
        <div className="ticker">
          {
            companies?.map((company) => (
              <p key={company.symbol} className="ticker__item">
                {`${company.symbol} : $${company.price}`}
              </p>
            ))
          }
        </div>
      </div>
    </div>
  );
};

export default Ticker;
