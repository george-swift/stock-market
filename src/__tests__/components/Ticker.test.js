import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import '../../setupTests';
import Ticker from '../../containers/Ticker';

describe('Screenshot of the Ticker component', () => {
  const mockStore = configureStore();
  const store = mockStore({
    listings: [
      {
        symbol: 'MU',
        name: 'Micron Technology, Inc.',
        price: 75.53,
        exchange: 'Nasdaq Global Select',
      },
      {
        symbol: 'LOW',
        name: "Lowe's Companies, Inc.",
        price: 196.77,
        exchange: 'New York Stock Exchange',
      },
    ],
  });

  it('should render correctly', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Ticker />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
