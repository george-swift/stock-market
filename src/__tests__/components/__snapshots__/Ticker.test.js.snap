// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screenshot of the Ticker component should render correctly 1`] = `ReactWrapper {}`;
